application/src/main/java/org/thingsboard/server/service/mail/TbMailSender.java

```diff
@@ -71,24 +70,27 @@ public TbMailSender(TbMailContextComponent ctx, JsonNode jsonConfig) {
         setJavaMailProperties(createJavaMailProperties(jsonConfig));
     }
 
-    @SneakyThrows
     @Override
     public void doSend(MimeMessage[] mimeMessages, @Nullable Object[] originalMessages) {
-        if (oauth2Enabled && (System.currentTimeMillis() > tokenExpires)){
-            refreshAccessToken();
-            setPassword(accessToken);
-        }
+        updateOauth2PasswordIfExpired();
         super.doSend(mimeMessages, originalMessages);
     }
 
-    @SneakyThrows
     @Override
-    public void testConnection() {
+    public void testConnection() throws MessagingException {
+        updateOauth2PasswordIfExpired();
+        super.testConnection();
+    }
+
+    private void updateOauth2PasswordIfExpired() {
         if (oauth2Enabled && (System.currentTimeMillis() > tokenExpires)){
-            refreshAccessToken();
+            try {
+                refreshAccessToken();
+            } catch (ThingsboardException e) {
+                throw new RuntimeException(e);
```

Comment:
The upper method is expected the `MessagingException` we are throwing RuntimeException
I'm not aware how the exception processing works on the upper level. Lets say the UI may show the human-readable short error. or weird long stack trace.
The short message is desired.
Please, verify from the UI side the difference between the error before and after your PR. It may have some difference. Add the screenshot of resulted UI error message to this PR description.